### PR TITLE
ignore error when fetching tags failed in ReadFunc

### LINKED ISSUE #331: raising an error message when refreshing vpc and subnet
```
terraform plan
  ----
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.huaweicloud_images_image_v2.ubuntu: Refreshing state...
huaweicloud_vpc_v1.test_vpc1: Refreshing state... [id=742e2332-5b6b-4891-be09-1aa4cbd05450]

Error: Error fetching HuaweiCloud VirtualPrivateCloud tags: Resource not found: [GET https://vpc.cn-north-1.myhuaweicloud.com/v2.0/e71eaa2d4efc4567abf35458e0f504da/vpcs/742e2332-5b6b-4891-be09-1aa4cbd05450/tags], error message: {"error_msg":"The API does not exist or has not been published in the environment","error_code":"APIGW.0101","request_id":"85015c404347baa2e10870b5ac4dc92e"}
```
introduced by #315 as some regions have not  published the tags.

### PR DESCRIPTION
**Pull request by ShiChangkuo**
fixes #331

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_vpc_subnet_v1.go [hunk 1]
```diff
@@ -208,12 +208,16 @@ func resourceVpcSubnetV1Read(d *schema.ResourceData, meta interface{}) error {
...
 	resourceTags, err := tags.Get(vpcSubnetV2Client, "subnets", d.Id()).Extract()
 	if err != nil {
-		return fmt.Errorf("Error fetching HuaweiCloud VpcSubnet tags: %s", err)
-	}
-
-	tagmap := tagsToMap(resourceTags.Tags)
-	if err := d.Set("tags", tagmap); err != nil {
-		return fmt.Errorf("Error saving tags for HuaweiCloud VpcSubnet (%s): %s", d.Id(), err)
+		if _, ok := err.(golangsdk.ErrDefault404); ok {
+			log.Printf("[INFO] fetching Subnet %s tags failed: The API does not exist or has not been published", d.Id())
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_vpc_subnet_v1.go]: seems 404 clould also come out if the subnet is not exist or something, maybe just warn we got a 404 here is enough?

File: huaweicloud/resource_huaweicloud_vpc_v1.go [hunk 2]
```diff
@@ -144,12 +144,16 @@ func resourceVirtualPrivateCloudV1Read(d *schema.ResourceData, meta interface{})
...
 	resourceTags, err := tags.Get(vpcV2Client, "vpcs", d.Id()).Extract()
 	if err != nil {
-		return fmt.Errorf("Error fetching HuaweiCloud VirtualPrivateCloud tags: %s", err)
-	}
-
-	tagmap := tagsToMap(resourceTags.Tags)
-	if err := d.Set("tags", tagmap); err != nil {
-		return fmt.Errorf("Error saving tags for HuaweiCloud VirtualPrivateCloud (%s): %s", d.Id(), err)
+		if _, ok := err.(golangsdk.ErrDefault404); ok {
+			log.Printf("[INFO] fetching VPC %s tags failed: The API does not exist or has not been published", d.Id())
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_vpc_v1.go]: ditto.

File: huaweicloud/resource_huaweicloud_vpc_subnet_v1.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/resource_huaweicloud_vpc_subnet_v1.go]: if the subnet is not exist, an error will raise in subnet.Get(), so we don't have to check more.

Comment (niuzhenguo) [reply huaweicloud/resource_huaweicloud_vpc_subnet_v1.go]: or Maybe we can extract the real error message from the err object. 404 can be possible be other issues.

Comment (niuzhenguo) [reply huaweicloud/resource_huaweicloud_vpc_subnet_v1.go]: https://github.com/huaweicloud/golangsdk/blob/master/errors.go#L123 seems we format the err already.

Comment (ShiChangkuo) [reply huaweicloud/resource_huaweicloud_vpc_subnet_v1.go]: ok, I will pull a new patch. thanks

**Pull request merged**